_posts/2020-03-02-ios-deploys-super-weird-totally-normal.markdown

```diff
@@ -0,0 +1,113 @@
+---
+layout: epic
+title: "iOS Deploys: Super Weird And Totally Normal"
+date: 2020-03-02
+categories: [ios, deployment, continuous deployment, fastlane]
+author: ash
+---
+
+Software deploys! What a concept. You have some code running somewhere, and you need to get it running somewhere
+else. What could possibly go wrong? While web developers have become accustom to some _really slick_ deploy
+processes, iOS developers have to work within some very different constraints.
+
+Today I want to explore the differences between deploying iOS software and front-end/back-end web software. Some of
+these differences are inherent to how the code gets executed, and some of the differences are incidental to choices
+that Apple has made. These are constraints that iOS developers need to work within. As Artsy has adopted React
+Native over the pats four years, we have had more and more of our web engineering colleagues contributing to our
+iOS app. For these web engineers, getting familiar with the iOS deploy constraints is as important as getting to
+know Xcode and CocoaPods.
+
+<!-- more -->
+
+We're going to use a case study to frame today's discussion. Artsy's Mobile Experience team recently got a ticket
+from our Platform team. The nature of the ticket itself doesn't matter, but it involved a change we were making to
+our API. I fixed the bug and submitted a pull request, which quickly got merged. We tested the bug fix in our next
+beta deploy, and everything looked fine. Great so far.
+
+Artsy releases iOS software on a two-week release cadence, and this bug fix was scheduled to be released to users
+the following week. But our back-end team wanted to quickly deploy that API change, which would require the app bug
+fix to get released to users _first_.
+
+So... what to do?
+
+Let's actually pause for a moment and consider one of the assumptions in that last paragraph. If you're a web
+engineer, the idea of releasing only every two weeks might seem pretty strange! I mean, why not release
+continuously? For example, Artsy's website gets deployed to our staging environment after every merged pull
+request, and staging then gets promoted to production several times a day. This process is generally referred to as
+"continuous delivery", and
+[it has a lot of advantages](https://www.thoughtworks.com/insights/blog/case-continuous-delivery). That's why it's
+so common among web engineering teams. So why not use continuous delivery on iOS apps?
+
+There are two reasons we can't use continuous delivery on iOS. First, continuous delivery is only really possible
+when you control where the software gets executed (or, in the case of web front-ends, where the client-side code
+gets served from). Artsy controls our own servers, so we can delivery web software continuously. However, the Artsy
```

Comment:
typo `delivery` -> `deliver`. 

I had to read this a couple times through; maybe a small addendum would help? "...web software continuously—when a user refreshes the page after a deploy, they'll start seeing the newest version right away."